src/torchmetrics/classification/group_fairness.py

```diff
@@ -0,0 +1,257 @@
+
+    Example (preds is int tensor):
+        >>> from torchmetrics.classification import BinaryFairness
+        >>> target = torch.tensor([0, 1, 0, 1, 0, 1])
+        >>> preds = torch.tensor([0, 1, 0, 1, 0, 1])
+        >>> groups = torch.tensor([0, 1, 0, 1, 0, 1])
+        >>> metric = BinaryFairness(2)
+        >>> metric(preds, groups, target)
+        ({'DP_0_1': tensor(0.)}, {'EO_0_1': tensor(0.)})
+
+    Example (preds is float tensor):
+        >>> from torchmetrics.classification import BinaryFairness
+        >>> target = torch.tensor([0, 1, 0, 1, 0, 1])
+        >>> preds = torch.tensor([0.11, 0.84, 0.22, 0.73, 0.33, 0.92])
+        >>> groups = torch.tensor([0, 1, 0, 1, 0, 1])
+        >>> metric = BinaryFairness(2)
+        >>> metric(preds, groups, target)
+        ({'DP_0_1': tensor(0.)}, {'EO_0_1': tensor(0.)})
+    """
+    is_differentiable = False
+    higher_is_better = False
+    full_state_update: bool = False
+
+    def __init__(
+        self,
+        num_groups: int,
+        task: Literal["demographic_parity", "equal_opportunity", "all"] = "all",
+        threshold: float = 0.5,
+        ignore_index: Optional[int] = None,
+        validate_args: bool = True,
+        **kwargs: Any,
+    ) -> None:
+        """Initialize states and validate arguments."""
+        super().__init__()
+
+        if task not in ["demographic_parity", "equal_opportunity", "all"]:
+            raise ValueError(
+                f"Expected argument `task` to either be ``demographic_parity``,"
+                f"``equal_opportunity`` or ``all`` but got {task}."
+            )
+
+        if validate_args:
+            _binary_stat_scores_arg_validation(threshold, "global", ignore_index)
+
+        self.num_groups = num_groups
+        self.task = task
+        self.threshold = threshold
+        self.ignore_index = ignore_index
+        self.validate_args = validate_args
+
+        self._create_states(self.num_groups)
+
+    def update(self, preds: torch.Tensor, groups: torch.Tensor, target: Optional[torch.Tensor] = None) -> None:
+        """Update state with predictions, groups, and target.
+
+        Args:
+            preds: Tensor with predictions.
+            groups: Tensor with group identifiers. The group identifiers should be ``0, 1, ..., (num_groups - 1)``.
+            target: Tensor with true labels.
+        """
+        if self.task == "demographic_parity":
+            if target is not None:
+                rank_zero_warn("The task demographic_parity does not require a target.", UserWarning)
+            target = torch.zeros(preds.shape)
+
+        group_stats = _binary_groups_stat_scores(
+            preds, target, groups, self.num_groups, self.threshold, self.ignore_index, self.validate_args
+        )
+
+        self._update_states(group_stats)
+
+    def compute(
+        self,
+    ) -> Union[Dict[str, torch.Tensor], Tuple[Dict[str, torch.Tensor], Dict[str, torch.Tensor]]]:
```

Comment:
I'm unsure whether a ``Tuple``is the right choice when more than 3-4 group-fairness metrics are added. Maybe I should change this to a dict of dicts?